File: 01_ChangeNext/01_ChangeNext_OC/01_ChangeNext_OC/ImageCell.m
//
//  ImageCell.m
//  01_ChangeNext_OC
//
//  Created by Fengtf on 2016/11/22.
//  Copyright © 2016年 ftf. All rights reserved.
//

#import "ImageCell.h"
#import "UIImageView+WebCache.h"

@interface ImageCell()


@property (nonatomic,weak)UIImageView * imageView;

@end


@implementation ImageCell


-(instancetype)initWithFrame:(CGRect)frame
{
    if (self = [super initWithFrame:frame]) {
        UIImageView *imageView = [[UIImageView alloc]init];
        self.imageView = imageView;
        [self addSubview:imageView];
        imageView.layer.cornerRadius = 10;
        imageView.clipsToBounds = YES;
    }
    return self;
}

-(void)layoutSubviews
{
    [super layoutSubviews];

    self.imageView.frame = self.bounds;
}

-(void)setUrl:(NSString *)url
{
    _url = url;
    [self.imageView sd_setImageWithURL:[NSURL URLWithString:url]];
}

@end

File: 01_ChangeNext/01_ChangeNext_Swift/ImageCell.swift
//
//  ImageCell.swift
//  01_ChangeNext_Swift
//
//  Created by Fengtf on 2016/11/23.
//  Copyright © 2016年 ftf. All rights reserved.
//

import UIKit

class ImageCell: UICollectionViewCell {

    var url:String?{
        didSet {
            guard let uurl = URL(string: url!) else {
                return
            }
            imageView.sd_setImage(with: uurl)
        }
    }

    private let imageView = UIImageView()

    override init(frame: CGRect) {
        super.init(frame: frame)


        self.addSubview(imageView)
        imageView.layer.cornerRadius = 10
        imageView.clipsToBounds = true
    }

    override func layoutSubviews() {
        super.layoutSubviews()

        imageView.frame = self.bounds
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
}
